ServerCore/Pages/Teams/Hints.cshtml.cs

```diff
@@ -47,19 +47,24 @@ public async Task<IActionResult> OnGetAsync(int puzzleId, int teamId)
             return Page();
         }
 
-        private async Task PopulateUI(int puzzleId, int teamId)
+        private async Task<IList<HintWithState>> GetAllHints(int puzzleId, int teamId)
```

Comment:
Both suggestions sound reasonable. There's a small problem though. I can't compute the base and adjusted costs in the HWS constructor because it requires the discount value which I can't compute until I have all the hints loaded. So I can create the hws-list as it does now, then compute the discount, then iterate over the list and call hws.SetDiscount() on each of them. Don't know what your coding conventions are here and if that breaks them. 

Let me know and I'll make the change.

Reply:
Personally, I'm totally fine with that approach, though Morgan has schooled me on coding style on more than one occasion and I'd like his take. :-)

Reply:
Yes, creating the list and then modifying it sounds fine. You could add AdjustedCost as a property of HintWithState. It would help avoid having too much logic in the display layer.

Reply:
I've refactored the code as suggested. HintWithState now has both BaseCost and AdjsutedCost properties which are used instead of hws.Hint.Cost.